### PR TITLE
MM-53460: Partition by account_external_id

### PR DESCRIPTION
**Pull request by ifoukarakis**
#### Summary

Properly deduplicate accounts to be updated.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-53460

### CODE REVIEW

**Review by sri-byte — APPROVED** [review]

File: transform/snowflake-dbt/models/hightouch/freemium/professional_subscription/freemium_account_update.sql [hunk 1]
```diff
@@ -43,43 +43,52 @@ WITH customers_with_cloud_paid_subs as (
...
+        domain_account_id,
+        domain_account_external_id,
+        -- Data to be updated
+        -- Select account id based on priority:
+        --  1. external id match
+        --  2. account from contact
+        --  3. account from lead
+        --  4. account with matching domain
+        case
+            when has_external_id_match then 'existing external id'
```
Comment (catalintomai) [inline transform/snowflake-dbt/models/hightouch/freemium/professional_subscription/freemium_account_update.sql:65]: @ifoukarakis - do you know where `has_external_id_match` is defined?

Comment (ifoukarakis) [reply transform/snowflake-dbt/models/hightouch/freemium/professional_subscription/freemium_account_update.sql:65]: Nice catch. Seems this was an old name and I copy-pasted from some old part of the code. Fixing it in hotfix PR.

Comment (catalintomai) [reply transform/snowflake-dbt/models/hightouch/freemium/professional_subscription/freemium_account_update.sql:65]: wondering if there is a tool out-there to catch these kind of simple referential errors.

Comment (ifoukarakis) [reply transform/snowflake-dbt/models/hightouch/freemium/professional_subscription/freemium_account_update.sql:65]: Best option right now seems to be DBT's slim CI

**Pull request merged**